internal/proxy/proxy_test.go

```diff
@@ -459,3 +461,64 @@ func TestClientInitializationWorksRepeatedly(t *testing.T) {
 	}
 	c.Close()
 }
+
+type spyHandler struct {
+	once sync.Once
+	done chan struct{}
+}
+
+func (s *spyHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	s.once.Do(func() { close(s.done) })
+	res.WriteHeader(http.StatusNotImplemented)
+}
+
+func (s *spyHandler) wasCalled() bool {
+	select {
+	case <-s.done:
+		return true
+	default:
+		return false
+	}
+}
+
+func TestClientInitializationWithCustomHost(t *testing.T) {
```

Comment:
We'll need to add this to avoid the underlying client from looking for valid credentials.

``` golang
	if testing.Short() {
		t.Skip("skipping Gcloud auth test")
	}
```